src/renderer/psp/js/widgets.js

```diff
@@ -0,0 +1,291 @@
+/*!
+Copyright 2017 Raising the Floor - International
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+/* global fluid */
+
+"use strict";
+(function (fluid) {
+    var gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.psp.widgets");
+
+    /**
+     * XXX currently, we abuse a misbehavior of expanding the `model` options, even if there's been expansion
+     * Manual expansion of the attrs is needed, as this "misbehaviour" only applies for the `model`
+     */
+    fluid.defaults("gpii.psp.widgets.attrsExpander", {
+        gradeName: "fluid.component",
+
+        // in case property is not passed, ensure
+        // it has won't affect the merged `attrs` property
+        rawAttrs: {},
+
+        // Currently if the used IoC expression string is passed directly to the `attrs`
+        // (see `gpii.psp.settingsPresenter` under the `widget` subcomponent), the expression
+        // is left unexpanded, thus resulting in improper `attrs` state.
+        // Ensure `attrs` receives expanded options properly through the `rawAttrs` property
+        // before merging takes place (FLUID-6219)
+        distributeOptions: {
+            target: "{that}.options.attrs",
+            record: "@expand:fluid.expandOptions({that}.options.rawAttrs, {that})"
+        }
+    });
+
+    fluid.defaults("gpii.psp.widgets.dropdown", {
+        gradeNames: ["gpii.psp.widgets.attrsExpander", "fluid.rendererComponent"],
+        model: {
+            optionNames: [],
+            optionList: [],
+            selection: null
+        },
+        modelListeners: {
+            "*": {
```

Comment:
To avoid unnecessary notifications, this path should be "" rather than "*" - https://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#wildcards-in-model-path-references